packages/gestalt-datepicker/src/DateRange.js

```diff
@@ -0,0 +1,285 @@
+          <Flex alignItems="start" justifyContent="center" direction="column">
+            <div className={borderStyles.dateFieldSection}>
+              <Flex gap={3}>
+                <Box width={isMobile ? MOBILE_DATEFIELD_WIDTH : DATEFIELD_WIDTH}>
+                  <InternalDateField
+                    autoComplete="off"
+                    mobileEnterKeyHint="enter"
+                    id={`datefield-start-${componentId}`}
+                    localeData={localeData}
+                    onChange={({ value }) => {
+                      if (value?.getTime() || value === null) onStartDateChange({ value });
+                    }}
+                    onError={onStartDateError}
+                    onBlur={onStartDateBlur}
+                    onFocus={onStartDateFocus}
+                    value={startDateValue}
+                    minDate={minDate}
+                    maxDate={maxDate}
+                    errorMessage={startDateErrorMessage}
+                  />
+                </Box>
+                <Box display="flex" height="100%" alignItems="center">
+                  <Text>—</Text>
+                </Box>
+                <Box width={isMobile ? MOBILE_DATEFIELD_WIDTH : DATEFIELD_WIDTH}>
+                  <InternalDateField
+                    autoComplete="off"
+                    mobileEnterKeyHint="enter"
+                    id={`datefield-end-${componentId}`}
+                    localeData={localeData}
+                    onChange={({ value }) => {
+                      if (value?.getTime() || value === null) onEndDateChange({ value });
+                    }}
+                    value={endDateValue}
+                    minDate={startDateValue}
+                    onError={({ errorMessage, value }) => {
+                      onEndDateError({ errorMessage, value });
+                    }}
+                    onBlur={onEndDateBlur}
+                    onFocus={onEndDateFocus}
+                    maxDate={maxDate}
+                    errorMessage={endDateErrorMessage}
+                  />
+                </Box>
+              </Flex>
+            </div>
+            <Box
+              width={isMobile ? '100%' : 629}
+              display={isMobile ? 'flex' : undefined}
+              justifyContent={isMobile ? 'center' : undefined}
+            >
+              <InternalDatePicker
+                localeData={localeData}
+                rangeStartDate={startDateValue}
+                rangeEndDate={endDateValue}
+                id={`datepicker-${componentId}`}
+                onChange={({ startDate, endDate }) => {
+                  onStartDateChange({ value: startDate });
+                  onEndDateChange({ value: endDate });
+                }}
+                minDate={minDate}
+                maxDate={maxDate}
+              />
+            </Box>
+            <Flex.Item alignSelf={isMobile ? 'center' : 'end'}>
+              <ButtonGroup>
+                <Box marginBottom={4} marginEnd={4}>
```

Comment:
We may not need to wrap in Box anymore in the ButtonGroup since it's adding extra spacing?

Reply:
DONE